docs/latest/design/egctl.md

```diff
@@ -0,0 +1,64 @@
+# Introduce egctl
+
+## Motivation
+
+EG should provide a command line tool with following capabilities:
+
+- One-step installation/uninstallation experience
+- Collect configuration from data plane
+- Update admin options such as changing log level
+- Analyse system configuration to diagnose any issues in envoy-gateway
+
+This tool is named `egctl`.
+
+## Syntax
+
+Use the following syntax to run `egctl` commands from your terminal window:
+
+```console
+egctl [command] [subcommand] [name] [flags]
+```
+
+where `command`, `name`, and `flags` are:
+
+* `command`: Specifies the operation that you want to perform on one or more resources,
+  for example `install`, `uninstall`, `proxy-config`, `version`.
+
+* `name`: Specifies the name of the specified pod. 
+
+* `flags`: Specifies optional flags. For example, you can use the `-f` or `--file` flags to specify the values for installing.
+
+If you need help, run `kubectl help` from the terminal window.
+
+## Operation
+
+The following table includes short descriptions and the general syntax for all the `egctl` operations:
+
+| Operation      | Syntax                          | Description                                               |
+| -------------- |---------------------------------|-----------------------------------------------------------|
+| `version`      | `egctl version`                 | Prints out build version information.                     |
+| `install`      | `egctl install -f FILENAME`     | Install or reconfigure EG on a cluster.                   |
+| `uninstall`    | `egctl uninstall`               | Uninstall EG from a cluster                               |
+| `proxy-config` | `egctl proxy-config SUBCOMMAND` | Retrieve information about proxy configuration from envoy |
+| `analyze`      | `egctl analyze`                 | Analyze configuration and print validation messages       |
+
+## Examples
+
+Use the following set of examples to help you familiarize yourself with running the commonly used `egctl` operations:
+
+```console
+# Install EG using the definition in custom-values.yaml
+egctl install -f custom-values.yaml
+
+# Uninstall EG
+egctl uninstall
```

Comment:
Thinking about future requirements, we should consider an `import` subcommand to start managing unmanaged proxy infra.

Reply:
what's `unmanaged proxy infra` mean, can you share more details?

Reply:
Unmanaged means that a user may have Envoy proxies that are not managed by EG. `egctl import` would pull those proxies into EG's management. The details are TBD, I just want to throw out ideas for us to consider during the initial design.

Reply:
we can update `import` later.